lib/utils.js

```diff
@@ -30,4 +32,16 @@ const getAgentInfo = () => ({
   name: pjson.name,
 });
 
-module.exports = { getCodeRef, getAgentInfo };
+const parseStringToArray = (str) => {
```

Comment:
I think it is better to call this utility as `parseAttributes`, as they can be an array or string. It will help to encapsulate the logic of parsing attributes inside of this utility.
In this case `str` parameter should be rename to `attributes`.